spec/ParseQuery.spec.js

```diff
@@ -5275,4 +5278,33 @@ describe('Parse.Query testing', () => {
     // Validate
     expect(result.executionStats).not.toBeUndefined();
   });
+
+  it('Query fails when wrapped within eachBatch and direct access is set to true', async () => {
+    await reconfigureServer({
+      directAccess: true,
+    });
+
+    Parse.CoreManager.setRESTController(
+      ParseServerRESTController(Parse.applicationId, ParseServer.promiseRouter({ appId: Parse.applicationId }))
+    );
```

Comment:
Why is this needed? The REST controller is set automatically, depending on `directAccess`.

Reply:
This is needed because `reconfigureServer`

https://github.com/parse-community/parse-server/blob/435f0d1a4239bd54ba3146a2f3b2c2a41be750ee/spec/helper.js#L190

overrides the actual parse server impelemntation which prevented the bug from being reproduced.

https://github.com/parse-community/parse-server/blob/435f0d1a4239bd54ba3146a2f3b2c2a41be750ee/src/ParseServer.js#L323-L325

Reply:
Got it